firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/FirebaseCrashlyticsReportManager.java

```diff
@@ -98,8 +98,14 @@ public void onCustomKey(String key, String value) {
     reportMetadata.setCustomKey(key, value);
   }
 
+  @Override
+  public void onUserId(String userId) {
+    reportMetadata.setUserId(userId);
+  }
+
   @Override
   public void onEndSession() {
+    reportPersistence.persistUserIdForSession(reportMetadata.getUserId(), currentSessionId);
```

Comment:
These two lines are fine! -- but I think this is where there's an opportunity to instead of calling reportPersistence and passing in a currentSessionId, making a per session-id object that gets persisted by these calls.

Reply:
Yup, let's chat more about this.